### PR TITLE
docs: Fix encapsulation decorators example

### PR DESCRIPTION
**Pull request by anthonyringoet**
Encapsulation documentation uses decorators (`foo`, `bar`) that are available on different levels but uses all of these at these different levels.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by RafaelGSS — APPROVED** [review]
Good Catch.

File: docs/Encapsulation.md [hunk 1]
```diff
@@ -51,9 +51,7 @@ fastify.register(async function authenticatedContext (childServer) {
     method: 'GET',
     handler (request, response) {
       response.send({
-        answer: request.answer,
-        foo: request.foo,
-        bar: request.bar
+        answer: request.answer
```
Comment (jsumners) [inline docs/Encapsulation.md:54]: The point of including the decorators here is to show that they are not available and will return as `undefined`.

File: docs/Encapsulation.md [hunk 2]
```diff
@@ -68,8 +66,7 @@ fastify.register(async function publicContext (childServer) {
     handler (request, response) {
       response.send({
         answer: request.answer,
-        foo: request.foo,
-        bar: request.bar
+        foo: request.foo
```
Comment (jsumners) [inline docs/Encapsulation.md:69]: Similarly, this is to show that `request.foo` is now available and will not be `undefined`, but `request.bar` _will_ still be `undefined`.

**Review by jsumners — COMMENTED** [review]
These examples are actually correct. Follow through with the full example to see that the intention is to show these decorators will return `undefined`.

**Comment by anthonyringoet** [comment]
Ah, my bad!

**Comment by jsumners** [comment]
Is there a change that would make this clearer?

**Comment by anthonyringoet** [comment]
@jsumners I think adding a couple of comments could make this more clear. Happy to push another change.

Example for the root context:
```javascript
{
  answer: request.answer
  // request.foo will be undefined as it's only defined in publicContext
  foo: request.foo,
  // request.bar will be undefined as it's only defined in grandchildContext
  bar: request.bar,
}
````

**Comment by jsumners** [comment]
That seems reasonable.

**Comment by anthonyringoet** [comment]
Updated

**Review by jsumners — APPROVED** [review]
LGTM

**Pull request merged**